### PR TITLE
snapguidist configuration

### PR DESCRIPTION
**Pull request by cef62**
This PR introduce a breaking change

### CODE REVIEW

File: src/context/runTest.js [hunk 1]
```diff
@@ -4,8 +4,10 @@ const headers = { 'Content-Type': 'application/json' }
 
 export default function runTest(snapshot, update) {
   const {
-    host = 'localhost',
-    port = 3000,
+    serverInfo: {
+      host = 'localhost',
```
Comment (MicheleBertoli) [inline src/context/runTest.js]: Aren't we setting the default values already in `index.js`?

Comment (cef62) [inline src/context/runTest.js]: yep we do, it can be removed 👍

Comment (cef62) [inline src/context/runTest.js]: But I've a question about the removal. In `runTest.spec` we execute the module outside webpack and passing a deeply structured object manually to `process.env.SNAPGUIDIST ` doesn't works.

This:
```javascript
process.env.SNAPGUIDIST = { serverInfo: { host: 'localhost', port: 3000 } }
``` 

returns `'[object Object]'`

This pass the contents but the `runTest` module receive it as a string and not an object.
```javascript
process.env.SNAPGUIDIST = JSON.stringify({ serverInfo: { host: 'localhost', port: 3000 } })
``` 

Do you have any idea on how to fix that situation?

Comment (MicheleBertoli) [inline src/context/runTest.js]: What if we run the test using a real env var?

Comment (cef62) [inline src/context/runTest.js]: Good idea! I fear though that without webpack `DefinePlugin` defining deep-object env variables could be problematic... I've to try it.
An other possible solution could be to use [jest globals](http://facebook.github.io/jest/docs/configuration.html#globals-object) to create the `process.env.SNAPGUIDIST` object.

Comment (MicheleBertoli) [inline src/context/runTest.js]: Oh yes, using jest globals makes a lot of sense. It's my preferred solution if it works. Thanks!

Comment (cef62) [inline src/context/runTest.js]: yes! using `jest globals` it worked!

**Comment by cef62** [comment]
@MicheleBertoli if the branch is ok for you now I'll rebase it before merging it to `master`.
Let me know if instead you prefer other changes.

File: src/context/runTest.js (hunk 1 above)
Comment (MicheleBertoli) [inline src/context/runTest.js]: Perfect! I guess we can remove it from this module then.

File: src/index.js [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
-      updateWebpackConfig(webpackConfig, env) {
-        let final = updateWebpackConfig(webpackConfig, env, serverInfo)
-        if (_updateWebpackConfig) {
-          final = _updateWebpackConfig(final, env)
-        }
-        return final
+function snapguidist(options = SNAPGUIDIST_DEFAULTS) {
+  return (config = {}) => {
+    const snapguidistOptions = Object.assign({}, options, {
+      serverInfo: {
```
Comment (MicheleBertoli) [inline src/index.js]: Can't we move this to the `SNAPGUIDIST_DEFAULTS`?

**Comment by MicheleBertoli** [comment]
@cef62 yup, just a couple of comments.
Thanks!

**Comment by cef62** [comment]
@MicheleBertoli I've updated the definition of the snapguidistOptions object.

**Comment by MicheleBertoli** [comment]
Perfect @cef62, thanks!

It's great to use `serverHost` and `serverPort` from the original config when they are present.
Why don't we just name our internal variables in the same way? It would make the config more consistent and easier to override.

Like:
```js
const SNAPGUIDIST_DEFAULTS = {
  concurrentTests: 3,
  serverHost: 'localhost',
  serverPort: 3000,
}
```

Please don't forget to remove the default values from `src/context/runTest.js` as well.

**Comment by cef62** [comment]
Ok! I'll flatten the default configuration object :)

**Comment by cef62** [comment]
I'm closing this PR and opening a new one to `master`

**Pull request closed without merging**